[simd] Add new port

Edi61 opened the pull request:
**Describe the pull request**
Created new port for https://github.com/ermig1979/Simd
related also to issue https://github.com/microsoft/vcpkg/issues/15008

- #### What does your PR fix?
No fix, introduction of a new port

- #### Which triplets are supported/not supported? Have you updated the [CI baseline]
Windows, Linux. Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes, maintenace guide is checked e.g. replaced vcpkg_build_msbuild by vcpkg_install_msbuild

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

dtmen committed:
Added new port for Simd

Created new port for Simd from
https://github.com/ermig1979/Simd

dtmen committed:
Added new port for Simd

Created/updated new port for Simd from
https://github.com/ermig1979/Simd

Edi61 committed:
Merge branch 'microsoft:master' into master

dtmen committed:
Merge branch 'master' of https://github.com/erdogan61/vcpkg

Edi61 committed:
Revert "Merge branch 'master' of https://github.com/erdogan61/vcpkg"

This reverts commit 789c93e81a6ea996649e314eca852180035361ef, reversing
changes made to c87a8183bd518b77b4092bfc27d24ad35686d4d7.

Edi61 committed:
portfile.cmake fixes

Update x64-windows-static* and fixes

Edi61 committed:
Update simd.json

Edi61 committed:
Added missing define

Set "#define in SIMD_STATIC/src/Simd/SimdConfig.h"

Edi61 committed:
Update simd.json

Edi61 committed:
Update simd.json

Edi61 committed:
Revert "Revert "Merge branch 'master' of https://github.com/erdogan61/vcpkg""

This reverts commit a97bdb05f4cd0590002ca953fbd158cab8db045c.

erdogan61 commented:
@microsoft-github-policy-service agree

erdogan61 commented on scripts/ci.baseline.txt:
```diff
@@ -987,6 +987,11 @@ shogun:x86-windows        = skip
+simd:arm-uwp=fail
+simd:arm64-windows=fail
+simd:x64-uwp=fail
+simd:x64-linux=fail
```
I don't have linux and also not have so much experience about cmake, therefore I have let it open for now. But maybe you can give me some idea how I can test it on a windows system?

erdogan61 replied:
I will try with Ununtu Wsl and provide a solution.

Edi61 committed:
Added Linux support

Edi61 committed:
Update simd.json

Edi61 committed:
Added Cmake install patch

Otherwise build error:
"ninja: error: unknown target 'install'"

Edi61 committed:
Update simd.json

Adela0814 commented on scripts/ci.baseline.txt:
```diff
@@ -987,6 +987,10 @@ shogun:x86-windows        = skip
 signalrclient:x64-uwp=fail
 signalrclient:arm-uwp=fail
 simbody:arm64-windows=fail
+simd:arm-uwp=fail
+simd:arm64-windows=fail
+simd:x64-uwp=fail
+simd:x64-osx=fail
```
Does `simd` really not support these triplets? I know you have no corresponding machine. Can you remove this and run again on CI?

erdogan61 replied:
->updated supports field in simd.json.

Adela0814 replied:
You should remove the unsupported from baseline.

Edi61 committed:
Added supports in vcpkg.json

Edi61 committed:
Update simd.json

Edi61 committed:
Removed unsupported builds

Edi61 committed:
x64_uwp failes

UWP builds are failing, since not supported,
Reason:
The following DLLs do not have the App Container bit set:

    D:/packages/simd_x64-uwp/debug/bin/Simd.dll
    D:/packages/simd_x64-uwp/bin/Simd.dll

Edi61 committed:
Update vcpkg.json

Edi61 committed:
Update simd.json

Edi61 committed:
Try ARM support, OSX not running

Osx build failed, arm maybe supported

Edi61 committed:
Update simd.json

Edi61 committed:
Fix syntax error and optimize

Edi61 committed:
Update simd.json

Edi61 committed:
Fix syntax

Edi61 committed:
Update simd.json

erdogan61 commented:
Removed all mentioned builds from Ci-Baseline, but uwb and osx failed since not supported. Arm seem to be supported: I have tried some update,
But i cannot access the log files from Azure, is there some possibility?

Edi61 committed:
Added SIMD_TARGET="aarch64"

Edi61 committed:
Update simd.json

Edi61 committed:
Seperated arm64 cmake

Check failes:
#if defined(_MSC_VER) && !defined(__clang__) && (defined(SIMD_X64_ENABLE) || defined(SIMD_X86_ENABLE))

#define SIMD_INIT_AS_CHAR

#elif defined(__GNUC__) || defined(__clang__) || (defined(_MSC_VER) && defined(SIMD_NEON_ENABLE))

#define SIMD_INIT_AS_LONGLONG

#else

#error This platform is unsupported!

#endif

Edi61 committed:
Update simd.json

Edi61 committed:
Syntax Error

Edi61 committed:
Update simd.json

Edi61 committed:
Update to latest release 5.2.120

*fix bug: MSVS compiler bug (Windows, Arm64).
commit b39197e00a902cbbc5ad397de92f086be994549b

Edi61 committed:
Baseline Updates

Edi61 committed:
Include arm64 fix

Edi61 committed:
Update SHA512

Edi61 committed:
Update simd.json

Edi61 committed:
Arm64 with msvc compiler not supported

Edi61 committed:
Update simd.json

Edi61 committed:
Updated supports field

Edi61 committed:
Baseline update

erdogan61 commented:
To sum up:
- uwb and osx are not supported
- arm* is supported generally (therefore added the support), but arm64 with MSVC compiler is not (-> CI build will fail)
Updated also supports field in vcpkg.json

Adela0814 commented on scripts/ci.baseline.txt:
```diff
@@ -987,6 +987,10 @@ shogun:x86-windows        = skip
 signalrclient:x64-uwp=fail
 signalrclient:arm-uwp=fail
 simbody:arm64-windows=fail
+simd:arm-uwp=fail
+simd:x64-uwp=fail
+simd:x64-osx=fail
+simd:arm64-windows=fail
```
This is unnecessary. You can delete it because it is added to the supports.

Edi61 committed:
Update ci.baseline.txt

Edi61 committed:
Github action fix

Replaced vcpkg_configure_cmake by vcpkg_cmake_configure

Edi61 committed:
Update simd.json
